blog/_posts/2018-05-02-developing-on-kubernetes.md

```diff
@@ -0,0 +1,366 @@
+Next, we had to patch `stock-con/app.yaml` slightly to make it work with Skaffold:
+
+Add a `namespace` field to both the `stock-con` deployment and the service with the value of `dok`.
+Change the `image` field of the container spec to `quay.io/mhausenblas/stock-con` since Skaffold manages the container image tag on the fly.
+
+ The resulting `app.yaml` file stock-con looks as follows:
+
+```
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app: stock-con
+  name: stock-con
+  namespace: dok
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: stock-con
+    spec:
+      containers:
+      - name: stock-con
+        image: quay.io/mhausenblas/stock-con
+        env:
+        - name: DOK_STOCKGEN_HOSTNAME
+          value: stock-gen
+        - name: DOK_STOCKGEN_PORT
+          value: "9999"
+        ports:
+        - containerPort: 9898
+          protocol: TCP
+        livenessProbe:
+          initialDelaySeconds: 2
+          periodSeconds: 5
+          httpGet:
+            path: /healthz
+            port: 9898
+        readinessProbe:
+          initialDelaySeconds: 2
+          periodSeconds: 5
+          httpGet:
+            path: /healthz
+            port: 9898
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: stock-con
+  name: stock-con
+  namespace: dok
+spec:
+  type: ClusterIP
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: 9898
+  selector:
+    app: stock-con
+```
+
+The final step before we can start development is to configure Skaffold. So, create a file `skaffold.yaml` in the `stock-con/` directory with the following content:
+
+```
+apiVersion: skaffold/v1alpha2
+kind: Config
+build:
+  artifacts:
+  - imageName: quay.io/mhausenblas/stock-con
+    workspace: .
+    docker: {}
+  local: {}
+deploy:
+  kubectl:
+    manifests:
+      - app.yaml
+```
+
+Now we’re ready to kick off the development. For that execute the following in the `stock-con/` directory:
+
+```
+$ skaffold dev
+```
+
+Above command triggers a build of the `stock-con` image and then a deployment. Once the pod of the `stock-con` deployment is running, we again forward the `stock-con` service for local consumption (in a separate terminal session) and check the response of the `healthz` endpoint:
+
+```bash
+$ kubectl get -n dok po --selector=app=stock-con  \
+                     -o=custom-columns=:metadata.name --no-headers |  \
+                     xargs -IPOD kubectl -n dok port-forward POD 9898:9898 &
+$ watch curl localhost:9898/healthz
+```
+
+If you now change the code in the `stock-con`directory, for example, by updating the []/healthz endpoint code in service.js](https://github.com/kubernauts/dok-example-us/blob/2334ee8fb11f8813370122bd46285cf45bdd4c48/stock-con/service.js#L52) by adding a field to the JSON response, you should see Skaffold noticing the change and create a new image as well as deploy it. The resulting screen would look something like this:
```

Comment:
pls check the link

Reply:
See above. It's a permalink and it works as designed.

Reply:
this is the same one as above

Reply:
Fixed